src/token.ts

```diff
@@ -35,7 +46,14 @@ export function pluginToken(provider: TokenProvider): ZodiosPlugin {
             error.config
           ) {
             if (error.response?.status === 401) {
-              const newToken = await provider.renewToken();
+              if (!pendingRenew) {
+                pendingRenew = provider.renewToken().then((token) => {
+                  pendingRenew = undefined;
```

Comment:
Don't this reset the pending promise token before the user get it back? Then below, newToken will be undefined in some cases?

Reply:
I don't think so, not in its current form, because clearing this reference will only affect instances which are not already `await`-ing the promise.

It would be an issue if there were some other async call before awaiting the renew token. e.g.

```js
if (pendingRenew) {
  await doSomethingElse();
  const newToken = await pendingRenew; // now `pendingRenew` could be undefined
}
```

But your point is a good one. This does look a little inception-y and it seems like TS doesn't give us good hints in this case.  Maybe it's worth keeping the `isRenewPending` for clarity's sake